postgres_exporter/common/queries_nodemx.yml

```
@@ -30,24 +30,34 @@ ccp_nodemx_process:
 
 
 ccp_nodemx_mem:
-  query: "
-with d(key, val) as
-(select key, val from monitor.cgroup_setof_kv('memory.stat'))
+  query: "WITH d(key, val) as (SELECT key, val FROM monitor.cgroup_setof_kv('memory.stat'))
 SELECT
- monitor.kdapi_scalar_bigint('mem_request') as request,
-case when monitor.cgroup_scalar_bigint('memory.limit_in_bytes') = 9223372036854771712 then 0 else monitor.cgroup_scalar_bigint('memory.limit_in_bytes') end as limit,
- (select val from d where key='cache') as cache,
- (select val from d where key='rss') as rss,
- (select val from d where key='shmem') as shmem,
- (select val from d where key='mapped_file') as mapped_file,
- (select val from d where key='dirty') as dirty,
- (select val from d where key='active_anon') as active_anon,
- (select val from d where key='inactive_anon') as inactive_anon,
- (select val from d where key='active_file') as active_file,
- (select val from d where key='inactive_file') as inactive_file,
- monitor.cgroup_scalar_bigint('memory.usage_in_bytes') as usage_in_bytes,
- monitor.cgroup_scalar_bigint('memory.kmem.usage_in_bytes') as kmem_usage_in_bytes
-"
+  monitor.kdapi_scalar_bigint('mem_request') AS request,
+  CASE
+    WHEN monitor.cgroup_mode() = 'legacy' THEN
+        (CASE WHEN monitor.cgroup_scalar_bigint('memory.limit_in_bytes') = 9223372036854771712 THEN 0 ELSE monitor.cgroup_scalar_bigint('memory.limit_in_bytes') END)
+    ELSE
+        (CASE WHEN monitor.cgroup_scalar_bigint('memory.max') = 9223372036854775807 THEN 0 ELSE monitor.cgroup_scalar_bigint('memory.max') END)
```

🤔 The following is shorter but less clear maybe.

❓ These very-big numbers are not the same. Why's that? I assumed they were MAX_INT or something.

```suggestion
        COALESCE(NULLIF(monitor.cgroup_scalar_bigint('memory.limit_in_bytes'), 9223372036854771712), 0)
    ELSE
        COALESCE(NULLIF(monitor.cgroup_scalar_bigint('memory.max'), 9223372036854775807), 0)
```

---

I should find out the why. But, yeah, it looks like the default max differs between v1 and v2.

---

I still don't know they _why_, but the _what_ looks like cgroup v1 used [the maximum 64-bit signed integer, rounded to the nearest page](https://unix.stackexchange.com/questions/420906/what-is-the-value-for-the-cgroups-limit-in-bytes-if-the-memory-is-not-restricte), whereas v2 is using MAX_INT.

---

On the COALESCE vs CASE refactor -- I like your suggestion. My suspicion is that the CASE statements will be more accessible, but it depends on who's reading :man_shrugging: